core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -1577,4 +1583,14 @@ protected final void onChannelClosed(Channel channel) {
             }
         }
     }
+
+    /**
+     * Ensures this transport is still started / open
+     * @throws IllegalStateException if the transport is not started / open
+     */
+    protected final void ensureOpen() {
```

Comment:
I guess I'm missing something, but you're checking that we are indeed open; if we don't hold the lock here aren't we subject to time-of-check-time-of-use problems?